spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -276,8 +297,10 @@ public void startAckThread() {
 	}
 
 	protected void send(DatagramPacket packet) throws Exception {
-		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		DatagramSocket socket = getSocket();
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(getDestinationAddress());
```

Comment:
So, if we worry about the `destinationAddress` in the `DatagramPacket` why do we don't use it in case of `socketExpression`?
Actually I don't see difference between the `DatagramSocket` which is built on the `ackPort` and that one returned by the `socketExpression`. I can do the same logic in the `socketExpression` which we have here in the `UnicastSendingMessageHandler.getSocket()`.

Am I missing anything?